src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -151,6 +154,29 @@ export default App.extend({
 
     this.showList();
   },
+  subscribe() {
+    const channel = Radio.channel('ws');
+    const filterKey = this.getState().getType();
+    const flowStates = this.getState('flowStates');
+
+    channel.request('subscribe', this.collection.models, { filters: { [filterKey]: this.filters } });
+
+    this.listenTo(channel, 'message', ({ category, resource, payload }) => {
+      const modelResource = category === 'ResourceCreated' ? payload.resource : resource;
+      const model = Radio.request('entities', 'get:store', modelResource);
+
+      if (this.collection.get(model) || model.isFetching()) return;
+
+      model.fetch().then(() => {
```

Comment:
I think this'll be:

```
const include = [
      'program-action.program',
      'flow.program-flow.program',
    ].join()

const data = {
  fields: { flows: ['name', 'state'] },
  include,
};

model.fetch({ data }).then(...
```

Honestly that's probably enough logic to move it to a `Radio.request('entities', 'fetch:action..`

Or maybe just add the fields to `'fetch:actions:model'` and use that over model.fetch here.

Reply:
Great thanks, I just got this working in my local dev environment. 

For now I'm just doing the following:

```js
  // js/apps/patients/worklist/worklist_app.js

  _getWsFetchOptions(filterKey) {
    if (filterKey === 'flows') return {};

    return {
      data: {
        fields: { flows: ['name', 'state'] },
        include: ['program-action.program', 'flow.program-flow.program'].join(),
      },
    };
  },
  subscribe() {

    . . .

    const options = this._getWsFetchOptions(filterKey);

    model.fetch(options).then(() => {

       . . .

    })
  },
```

Reply:
Having this working now allows me to finish creating the initial tests for the action version of the worklist. After that I'll come back and do some refactoring on this and other parts too.